src/shims/foreign_items.rs

```diff
@@ -364,178 +315,41 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 this.write_scalar(Scalar::from_int(result, Size::from_bits(32)), dest)?;
             }
 
-            "memrchr" => {
+            "memchr" => {
```

Comment:
In the interest of keeping the diff smaller, please keep the original order: memrchr first, then memchr.